.buildkite/pipeline.yml

```diff
@@ -17,16 +28,18 @@ steps:
 
         buildkite-agent artifact upload bundle/android/App.js
 
+  - label: "Build Android RN Aztec & Publish to S3"
+    key: "publish-react-native-aztec-android"
+    <<: *publish-android-artifacts-docker-container
+    command: |
+        cd ./gutenberg/packages/react-native-aztec/android
+        ./gradlew -PwillPublishReactNativeAztecBinary=true :prepareToPublishToS3 `prepare_to_publish_to_s3_params` :publish
+        buildkite-agent meta-data set "PUBLISHED_REACT_NATIVE_AZTEC_ANDROID_VERSION" `cat ./build/published-version.txt`
+
   - label: "Build Android RN Bridge & Publish to S3"
-    depends_on: "bundle-android"
-    plugins:
-      - docker#v3.8.0:
-          image: "public.ecr.aws/automattic/android-build-image:v1.0.0"
-          propagate-environment: true
-          environment:
-            # DO NOT MANUALLY SET THESE VALUES!
-            # They are passed from the BuildKite agent to the Docker container
-            - "AWS_ACCESS_KEY"
-            - "AWS_SECRET_KEY"
+    depends_on:
+      - "bundle-android"
+      - "publish-react-native-aztec-android"
+    <<: *publish-android-artifacts-docker-container
```

Comment:
Completely opinionated but personally I prefer to avoid including the Map's key in the anchor (and then using the `<<:` YAML merge operator to include it), because it feels a bit less obvious to read at call site.

For example here at first glance I almost thought that the `*publish-android-artifacts-docker-container` line was part of the `depends_on` map key before realizing it was isntead merged with the parent key.

While if you only DRY the value of the Map's key, creating a YAML anchor only on the value not the whole 1-keypair map, sure you then have to repeat the key at call site, so I agree it's debatable in terms of DRY-ness, but it acts as a reminder about what this is about and helps readability in my humble opinion.

<details><summary>New way to define the anchor at the top of the YAML</summary>

```yaml
publish-android-artifacts-docker-container: &publish-android-artifacts-docker-container
  - docker#v3.8.0:
      image: "public.ecr.aws/automattic/android-build-image:v1.0.0"
      propagate-environment: true
      environment:
        # DO NOT MANUALLY SET THESE VALUES!
        # They are passed from the Buildkite agent to the Docker container
        - "AWS_ACCESS_KEY"
        - "AWS_SECRET_KEY"
```
</details>

Then these lines above would become this, which makes it clearer that the entry is setting plugins and is separate from depends_on:
```suggestion
    depends_on:
      - "bundle-android"
      - "publish-react-native-aztec-android"
    plugins: *publish-android-artifacts-docker-container
```

Again, this is completely opinionated, and is ultimately not a big deal and just about style, just figured I'd share it anyway to see what you think 🙂

Reply:
@AliSoftware This is actually exactly what I tried to do in https://github.com/wordpress-mobile/gutenberg-mobile/pull/3758/commits/afa3c2eea0186a927ea331c1fdb5f944de39abf8, but it had a parse error, so I changed it in https://github.com/wordpress-mobile/gutenberg-mobile/pull/3758/commits/312533a83d4b054f72a0c938b190a9d048678400.

I am very new to YAML anchors and I briefly tried to learn more about them, but I didn't find [the documentation](https://yaml.org/spec/1.2/spec.html) very approachable. So, I didn't want to spend too much time on it.

That's a very long way to say **thank you** for explaining the right way to do what I wanted in the first place! I'm opening a new PR shortly to address the feedback from you and @mokagio 🙇

Reply:
OK _exactly_ might not have been the correct term here, because this is actually what I'd have wanted so that we can add other plugins easily:

```
plugins:
    - *publish-android-artifacts-docker-container
```

or something similar with the correct YAML syntax?

Reply:
Yep you can do that as well that's a good idea. Simply move the & anchor between the `-` and `docker#v3.8.0` so that the anchor defines just that key not the 1-item-array containing it.

Actually you can even remove the `-` entirely in the definition, that would even be cleaner.

---

Basically when you define an 2anchor with `&foo` then YAML captures the node that is immediately after it.

So if you write:
```yaml
&foo
dictkey1:
  - an
  - array
dictkey2: value2
```
Then it will capture the whole dictionary (aka map) with its 2 keys. But if you write:
```yaml
dictkey1: &foo
  - an
  - array
dictkey2: value2
```
Then it will capture the value of the `dictkey1` key, which is the array of 2 items.

And if you write:
```yaml
dictkey1:
  - &foo an
  - array
dictkey2: value2
```
Then it will just capture the string `an`.

---

Note that in your YAML in this PR, the key that you use **before** your anchor is not really useful. We usually use one just in order to not put whatever we want to capture in the anchor… at the very root of the YAML, to not risk messing up with BuiltKite's official keys. So instead we define the anchor inside an arbitrary sub key to define it out of the way.

Instead of this, another way of doing it is to not define the anchor at the top of the YAML but instead at the first place of the node being used, and then use `*foo` for subsequent uses. But that feels a bit asymmetric and less easy to find definitions so that's why we add arbitrary keys at the top to define them separately

---

Another way to do it, which might make it clearer that the key we use before the anchor in these cases is arbitrary and unrelated to the anchor name, is to use a different key, and even why not group all the anchors we define under a single key with an array as values, one for each anchor. Tbh I find that more clear in intent. This is how I did it for example [here](https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/blob/develop/.circleci/config.yml#L7-L15) where I define 3 anchors, each being an item of an array inside an arbitrary-named key whose only purpose is to group those anchor definitions together (and to avoid defining each node we create an anchor for… at the top level, polluting BuiltKite's namespace). I think that last way of doing it also helps understanding anchor definitions better.

Reply:
@AliSoftware This whole explanation is extremely helpful, I really appreciate it.

_P.S: I've bookmarked this comment, so I can re-read it when I am working with YAML anchors again._ 🤗